metomi/rose/rose.py

```diff
@@ -14,11 +14,39 @@
 #
 # You should have received a copy of the GNU General Public License
 # along with Rose. If not, see <http://www.gnu.org/licenses/>.
+
+import os
+import sys
+
+
+def pythonpath_manip():
+    """Stop PYTHONPATH contaminating the Rose Environment
+
+    * Remove PYTHONPATH items from sys.path to prevent PYTHONPATH
+      contaminating the Rose Environment.
+    * Add items from ROSE_PYTHONPATH to sys.path.
+
+    See Also:
+        https://github.com/cylc/cylc-flow/issues/5124
+    """
+    if 'ROSE_PYTHONPATH' in os.environ:
+        for item in os.environ['ROSE_PYTHONPATH'].split(os.pathsep):
+            print(f'extracted {item} from ROSE_PYTHONPATH')
+            abspath = os.path.abspath(item)
+            sys.path.insert(0, abspath)
```

Comment:
Aware this is the same logic added in Cylc, but I've just noticed this reverses the order set by the user?

Reply:
Ach, dammit.

Reply:
```suggestion
        rose_pythonpath_vars = [
            os.path.abspath(item)
            for item in os.environ['ROSE_PYTHONPATH'].split(os.pathsep)
        ]
        print(f"Extracted {rose_pythonpath_vars} from ROSE_PYTHONPATH")
        sys.path.insert(0, *rose_pythonpath_vars)

```